pilot/pkg/networking/deprecated/clusters.go

```diff
@@ -0,0 +1,451 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package deprecated
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	_ "github.com/golang/glog" // nolint
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/duration"
+
+	alpha3 "istio.io/api/networking/v1alpha3"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/proxy/envoy/v1"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+
+	// OutboundClusterPrefix is the prefix for service clusters external to the proxy instance
+	OutboundClusterPrefix = "out."
+
+	// ClusterTypeStatic name for clusters of type 'static'
+	ClusterTypeStatic = "static"
+
+	// DefaultLbType defines the default load balancer policy
+	DefaultLbType = LbTypeRoundRobin
+
+	// LbTypeRoundRobin is the name for round-robin LB
+	LbTypeRoundRobin = "round_robin"
+
+	// ClusterFeatureHTTP2 is the feature to use HTTP/2 for a cluster
+	ClusterFeatureHTTP2 = "http2"
+
+	// ClusterTypeStrictDNS name for clusters of type 'strict_dns'
+	ClusterTypeStrictDNS = "strict_dns"
+
+	// ClusterTypeOriginalDST name for clusters of type 'original_dst'
+	ClusterTypeOriginalDST = "original_dst"
+
+	// ClusterTypeSDS name for clusters of type 'sds'
+	ClusterTypeSDS = "sds"
+
+	// LbTypeOriginalDST is the name for LB of original_dst
+	LbTypeOriginalDST = "original_dst_lb"
+)
+
+// BuildClusters2 returns the list of clusters for a proxy
+func BuildClusters2(env model.Environment, node model.Proxy) ([]*xdsapi.Cluster, error) {
+	clusters := []*xdsapi.Cluster{}
+	var proxyInstances []*model.ServiceInstance
+	var err error
+	switch node.Type {
+	case model.Sidecar, model.Router:
+		proxyInstances, err = env.GetProxyServiceInstances(node)
+		if err != nil {
+			return clusters, err
+		}
+		var services []*model.Service
+		services, err = env.Services()
+		if err != nil {
+			return clusters, err
+		}
+		clusters = buildSidecarClusters2(env.Mesh, proxyInstances,
```

Comment:
Please don't call functions "x2". This is not a maintainable style, it's unclear what the number means.